kuka_rsi_hw_interface/krl/README.md

```diff
@@ -1,4 +1,93 @@
+9. Run **cmd.exe** and ping the PC you want to communicate with on the same subnet (e.g. 192.168.250.xx).
+
+If your **PC** has an IP address on the same subnet as the **Windows interface** on the controller, the controller should receive answers from the PC:
+* If this is the case, add another IP address to the current PC connection (e.g. 192.168.1.xx) on the same subnet as the **RSI** interface.
+
+## 2. KRL Files
+
+The files included in this folder specifies the data transferred via RSI. Some of the files needs to be modified to work for your specific configuration.
+
+##### ros_rsi_ethernet.xml
+1. Edit the `IP_NUMBER` tag so that it corresponds to the IP address (192.168.1.xx) previously added for your PC.
+2. Keep the `PORT` tag as it is (49152) or change it if you want.
+
+##### ros_rsi.rsi.xml
+This file may be edited with application specific joint limits.
+* Edit the parameters within the RSIObject `AXISCORR` to specify joint limits such as **LowerLimA1, UpperLimA1** etc.
+* Edit the parameters within `AXISCORRMON` to specify the amount of degrees in which the robot should have reached its goal. The values of **MaxA1, MaxA2** etc. may be large to allow free movement within the specified joint limits in `AXISCORR`.
+
+Notice the RSIObject of type `ETHERNET`. Within this object is a parameter called **Timeout**. This parameter is set to **100** by default. The RSI interface operates at 250 Hz and expects new data every 4ms. If the connected **PC with ROS** does not fulfill this, a message is missed and a counter is incremented. When this counter hits **100**, the RSI connection will shut down. If you have problems with the connection to RSI shutting down now and then while moving the robot it is suggested to:
+* Compile and install a [RT-Preempt](https://rt.wiki.kernel.org/index.php/RT_PREEMPT_HOWTO) kernel for your PC.
+* Give **kuka_rsi_hardware_interface** on your PC real-time priority when the RSI connection is established.
+
+##### ros_rsi.src
+This should only be edited if the start position specified within the file is not desirable for your application.
+
+##### Copy files to controller
+The files **ros_rsi.rsi** and **ros_rsi.rsi.diagram** should not be edited. All files should now be ready to be copied to the Kuka controller:
+
+1. Copy the files to a USB-stick.
+2. Plug it into the teach pad or controller.
+3. Log in as **Expert** or **Administrator**.
+4. Copy the `ros_rsi.src` file to `KRC:\R1\Program`.
+5. Copy the rest of the files to `C:\KRC\ROBOTER\Config\User\Common\SensorInterface`.
+
+## 3. Edit kuka_rsi_hw_interface configuration files
+Inside `/kuka_rsi_hw_interface/test` in this repository is a set of `*.yaml` files. Ensure that these files have the correct **joint names** corresponding to the robot description (URDF or .xacro) and correct **IP address** and **port** corresponding to the RSI setup specified in **ros_rsi_ethernet.xml**.

```

Comment:
It would probably be preferable to have users *copy* the configuration files, and make their own launch file (based on a template we provide). That way, the same set of instructions will be usable when this package gets released (as that would make editing the files impossible (if they are even included in their current form in a `deb`)).

It would also immediately show them how to work with the package in the intended way (ie: not edit files in upstream packages, but create a derivative).